### DISCUSSION TITLE
How to call companion's apply

### DISCUSSION THREAD

**Discussion by Odomontois** [scala/scala3#15620] [Metaprogramming]
Suppose one have macros with 
```scala
def foo[A : Type](using Quotes): Expr[...] = 
```
At some points I already know that A is a `case class` and need to construct an expression of calling companion's `apply` method
How do I do this?
Currently I have

```scala

  val companion = aType.companionClass

  val apply = companion
      .declaredMethod("apply")
      .headOption
      .getOrElse(
        report.errorAndAbort(s"could not find apply method in $companion"),
      )
  
  val noPrefix = ???    
  val ref     = Select(Ident(TermRef(noPrefix , aType.name)), apply)
  
  Apply(ref, List(repr.asTerm)).asExpr.asExprOf[A]
 ```
 
 An just can't find out how to receive this `noPrefix` value of NoPrefix type, or how to reference the companion other way

**Comment by soronpo** [comment]
`companion` should be:

`val companion = TypeRepr.of[A].typeSymbol.companionModule`
